immuneML/reports/encoding_reports/DesignMatrixExporter.py

```diff
@@ -50,8 +61,34 @@ def _generate(self) -> ReportResult:
         return ReportResult(self.name, output_tables=[matrix_result], output_text=[details_result, label_result])
 
     def _export_matrix(self) -> ReportOutput:
+        """Create a file for the design matrix in the desired format."""
+        
         data = self._get_data()
-        file_path = self._save_to_file(data, self.result_path / "design_matrix")
+        file_path = self.result_path / "design_matrix"
+
+        # Use h5py to create a hdf5 file.
+        if self.format.endswith(("hdf5", "hdf5.zip")):
+            file_path = file_path.with_suffix(".hdf5")
+            with h5py.File(file_path, 'w') as hf_object:
+                hf_object.create_dataset(str(file_path), data=data)
+        # Use numpy to create a csv or npy file.
+        elif len(data.shape) <= 2 and self.format.endswith(("csv", "csv.zip")):
+            file_path = file_path.with_suffix(".csv")
+            np.savetxt(fname=str(file_path), X=data, delimiter=",", comments='',
+                       header=",".join(self.dataset.encoded_data.feature_names))
+        else:
+            if not self.format.endswith(("npy", "npy.zip")):
+                logging.info('The selected Report format is not compatible, '
+                             '.npy is used instead')
+            file_path = file_path.with_suffix(".npy")
+            np.save(str(file_path), data)
+        
+        # If requested, compress the file into a .zip.
+        if self.format.endswith(".zip"):
+            file_path_zip = str(file_path) + ".zip"
+            with zipfile.ZipFile(file_path_zip, 'w') as zipped_file:
+                zipped_file.write(file_path_zip, compress_type=zipfile.ZIP_DEFLATED)
+            os.remove(str(file_path))
         return ReportOutput(file_path, "design matrix")
```

Comment:
if `file_path` is removed here, then `file_path_zip` should be forwarded to the `ReportOutput` object. For instance, you could add a line to say:
```
file_path = file_path_zip
```
after you remove the `file_path` in line 91.

Reply:
done !